x-pack/plugins/transform/public/app/common/request.ts

```diff
@@ -134,44 +134,51 @@ export const getCreateTransformRequestBody = (
   indexPatternTitle: IndexPattern['title'],
   pivotState: StepDefineExposedState,
   transformDetailsState: StepDetailsExposedState
-): PutTransformsRequestSchema => ({
-  ...getPreviewTransformRequestBody(
-    indexPatternTitle,
-    getPivotQuery(pivotState.searchQuery),
-    dictionaryToArray(pivotState.groupByList),
-    dictionaryToArray(pivotState.aggList)
-  ),
-  // conditionally add optional description
-  ...(transformDetailsState.transformDescription !== ''
-    ? { description: transformDetailsState.transformDescription }
-    : {}),
-  // conditionally add optional frequency
-  ...(transformDetailsState.transformFrequency !== ''
-    ? { frequency: transformDetailsState.transformFrequency }
-    : {}),
-  dest: {
-    index: transformDetailsState.destinationIndex,
-  },
-  // conditionally add continuous mode config
-  ...(transformDetailsState.isContinuousModeEnabled
-    ? {
-        sync: {
-          time: {
-            field: transformDetailsState.continuousModeDateField,
-            delay: transformDetailsState.continuousModeDelay,
+): PutTransformsRequestSchema => {
+  const tempSettings = {
+    settings: {} as Record<string, number | null | undefined>,
+  };
+  if (transformDetailsState.transformSettingsMaxPageSearchSize) {
+    tempSettings.settings.max_page_search_size =
+      transformDetailsState.transformSettingsMaxPageSearchSize;
+  }
+  if (transformDetailsState.transformSettingsDocsPerSecond) {
+    tempSettings.settings.docs_per_second = transformDetailsState.transformSettingsDocsPerSecond;
+  }
```

Comment:
Suggest to move this to a function `getCreateTransformRequestBody` and use that further down after `// conditionally add additional settings` - similar pattern how `getPreviewTransformRequestBody()` is used. The function should have as return type `PutTransformsRequestSchema['settings']`. This should allow us to keep the existing structure of this `getCreateTransformRequestBody` function and hopefully will allow you to get rid of the temporary type `{} as Record<string, number | null | undefined>`.

Reply:
I have updated it here https://github.com/elastic/kibana/pull/78364/commits/10e30dd8a5e2d489eea85da10a1ffe59af6555b8

Although I ended up having to use the temporary type still because extending the typeof schema gives a Readonly type. 
```settings: Readonly<{max_page_search_size?: number | undefined, docs_per_second?: number | null | undefined}> | undefined```

Perhaps what I can do is defining the PutTransformsRequestSchema['settings'] a bit more explicitly instead of relying on the schema?